File: backend/chat-store-service/src/main/java/com/example/chatstoreservice/chat/dto/ChatType.java
package com.example.chatstoreservice.chat.dto;

import lombok.Getter;

import java.util.HashMap;
import java.util.Map;

@Getter
public enum ChatType {
    TEXT(1, ""),
    IMAGE(2, "image"),
    FILE(3, "file"),
    AUDIO(4, "audio");

    private final int chatType;
    private final String chatTypeStr;

    private static final Map<Integer, ChatType> chatTypeMap = new HashMap<>();

    static {
        for(ChatType chatType : values()) {
            chatTypeMap.put(chatType.getChatType(), chatType);
        }
    }

    ChatType(int chatType, String chatTypeStr) {
        this.chatType = chatType;
        this.chatTypeStr = chatTypeStr;
    }

    public static ChatType fromChatType(int type) {
        return chatTypeMap.get(type);
    }
}

File: backend/ws-service/src/main/kotlin/com/example/wsservice/chat/dto/ChatType.kt
package com.example.wsservice.chat.dto

enum class ChatType(val chatType: Int, val chatTypeStr: String) {
    TEXT(1, ""),
    IMAGE(2, "image"),
    FILE(3, "file"),
    AUDIO(4, "audio");

    companion object {
        private val byType = entries.associateBy { it.chatType }

        @JvmStatic
        fun fromChatType(type: Int): ChatType? = byType[type]
    }
}
